Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -797,12 +797,19 @@ open class UnsyncedBookmarksFallbackModelFactory: BookmarksModelFactory {
         log.debug("Getting model for fallback root.")
         // Return a virtual model containing "Desktop bookmarks" prepended to the local mobile bookmarks.
         return self.localFactory.folderForGUID(BookmarkRoots.MobileFolderGUID, title: BookmarksFolderTitleMobile)
-            >>== { folder in
-            return self.bufferFactory.getDesktopRoots() >>== { cursor in
-                let desktop = self.bufferFactory.folderForDesktopBookmarksCursor(cursor)
-                let prepended = PrependedBookmarkFolder(main: folder, prepend: desktop)
-                return deferMaybe(BookmarksModel(modelFactory: self, root: prepended))
-            }
+            >>== {
+                localMobileFolder in
+                
+                self.bufferFactory.folderForGUID(BookmarkRoots.MobileFolderGUID, title: BookmarksFolderTitleMobile) >>== {
+                    bufferMobileFolder in
+
+                    self.bufferFactory.getDesktopRoots() >>== { cursor in
+                        let bufferAndLocalMobile = ConcatenatedBookmarkFolder(main: bufferMobileFolder, prepend: localMobileFolder)
```

Comment:
You should really have local come after remote — the local items will be appended to the upstream folder when we upload them, so if you do it this way they'll reorder themselves after a sync, which is weird.

You can do that by just changing `prepend` to `append` and flipping some bits around in the implementation, which is probably clearest.